### ISSUE TITLE
Xcode iOS Export godot_path returning Invalid project path specified: "project_name", aborting.

### ISSUE THREAD

**Issue by yemibox51** [godotengine/godot#98474]
### Tested versions

Reproducible in 4.3.Stable.Official

### System information

macOS Sonoma, M2 Pro 16gb Ram

### Issue description

Followed the tutorial outlined here: https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_ios.html and Build succeeds but then I get this error:

```
Invalid project path specified: "project_name", aborting.
```

### Steps to reproduce

Use Godot 4.3 Stable on MacOS Sonoma with Xcode 16

Followed the tutorial outlined here: https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_ios.html.

In `Supporting Files`> `export_project.info-plist` added `godot_path` as key and the exact name of the referenced folder as the value in i.e. `project_name`.
Aware that the project name should different than the exported project file/folder

### Minimal reproduction project (MRP)

N/A

**Comment by syedmohsinbukhari** [comment]
I am having the exact same issue.

**Comment by yemibox51** [comment]
@syedmohsinbukhari unfortunately no. I manually re-export every time I make a change from Godot. It isn't very pleasant, but I am not totally blocked by this.

**Comment by gavyn-ezell** [comment]
Hoping it gets addressed! For now just exporting everytime

**Comment by syedmohsinbukhari** [comment]
@yemibox51 same here. I have been manually exporting as well. It is quite a nuisance. I hope this gets their attention.

**Comment by tevenson** [comment]
I'm running into the same thing with Godot 4.3.stable and Xcode 16.1.

**Comment by peterworth** [comment]
Same thing here, Godot 4.2 in my case, and Xcode 16.1

**Comment by smmathew** [comment]
Same for me

**Comment by redgriftjosh** [comment]
Same for me!

**Comment by kdaker** [comment]
This worked for me with an extra step:

Go to the Target Build Phases tab and add a "Copy Files" step. Select the referenced godot folder:

<img width="777" alt="Image" src="https://github.com/user-attachments/assets/4dace9ae-d5c0-48e6-882c-7b6c7117fb9d" />

**Comment by smmathew** [comment]
> This worked for me with an extra step:
> 
> Go to the Target Build Phases tab and add a "Copy Files" step. Select the referenced godot folder:
> <img alt="Image" width="777" src="https://private-user-images.githubusercontent.com/905997/399538826-4dace9ae-d5c0-48e6-882c-7b6c7117fb9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU3MzM0NDAsIm5iZiI6MTczNTczMzE0MCwicGF0aCI6Ii85MDU5OTcvMzk5NTM4ODI2LTRkYWNlOWFlLWQ1YzAtNDhlNi04ODJjLTdiNmM3MTE3ZmI5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwMVQxMjA1NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZDhhZGI0ZTA5MjliZGY4ZDJmNmQ3NmM1M2RlMWM2ZmUxOTA4ZmUwYjYzNjQ5NzE5MWEzOTAwMjljNDU0OWViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gO3KVm0pW1qcu3dg_7Jq2KkVYtERDpqWSBtLJ9afmBQ">

Do you have to select every file from that folder? I don't see a way to just choose an entire folder.

**Comment by kdaker** [comment]
@smmathew When you add the godot directory to xcode, make sure to set the Groups to use folders. Then in the inspector on the right hand side, set the build rules for that folder to "Apply once to folder".

<img width="652" alt="Image" src="https://github.com/user-attachments/assets/fb917393-75d3-4d0e-ac7f-beed60d3c6fd" />
<img width="256" alt="Image" src="https://github.com/user-attachments/assets/d77063fe-bfdf-497b-baf7-ee5c2cb201c7" />

**Comment by smmathew** [comment]
@kdaker I am still getting "Invalid project path specified: "jumper_jam", aborting." After godot loads on the phone after following your instructions. Any idea why it can't find the folder after I added it?<img width="267" alt="Image" src="https://github.com/user-attachments/assets/5bc9ddb0-1644-4714-9d1b-25135c8d9d76" />

**Comment by smmathew** [comment]
@kdaker I got it working! I had to go add the folder in the copy build phase. Thanks for your help the key I was missing was the "apply once to folder".

<img width="534" alt="Image" src="https://github.com/user-attachments/assets/0efb5681-cbf9-4eef-bad2-cd7f1e772894" />

**Comment by yemibox51** [comment]
Using this method significantly increases the size of the app. Has anyone found a way to decrease it without switching back to the `.pck`?

**Comment by alkhimey** [comment]
The method endorsed by the documentation requires copying the whole project directory into the bundle, this will include `.git` directory, `.godot` directory with all the build intermediate artifacts and any other hidden or irrelevant files you might have in the project directory. 

It takes time to copy and makes the app bundle huge. A starter mono project will have a size greater than 1GB (after exporting for release and debug once). Additionally one should not use this method when exporting to production but it is easy to forget or overlook this.

What I recommend is to add a build phase to the Xcode project to re-generate PCK for each build.

For mono projects, it is also required to copy `iOS-Game-Test_aot.xcframework`. The process that generates PCK already builds an xcframework for Release. There is no need to do it specifically for Debug as debugging is not possible anyway.

In Xcode, the commands should be added to a "build phase" that happens before linking.

```sh
GODO_PROJECT_ROOT=~/Documents/Projects/Godot-Experimentation/ios-game-test

/Applications/Godot_mono.app/Contents/MacOS/Godot --headless --path $GODO_PROJECT_ROOT --export-pack iOS ${PROJECT_DIR}/${PROJECT_NAME}.pck

cp -r $GODO_PROJECT_ROOT/.godot/mono/temp/bin/ExportRelease/iOS-Game-Test_aot.xcframework ${PROJECT_DIR}/${PROJECT_NAME}/dylibs/ExportDebug
```

![Image](https://github.com/user-attachments/assets/a690b90b-e827-43af-933e-a5b046b55237)

I found this method to be faster. Maybe for bigger projects it will become too slow eventually but for me it works so far. I believe that if needed, is possible to tell Xcode which files this build phase depends on and run the script only when something in the Godot project changes.

**Comment by dannys42** [comment]
I'm just starting with Godot, so I don't know whether it's better to have PCK files or the bundle.  But for anyone else trying to find the minimal steps needed to go from the [documentation](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_ios.html) to a working iOS build, this is what worked for me after following some of the steps people mentioned above:

- Drag the godot project into your Xcode project workspace
    -  Action: Reference files in place
    - Groups: Create folders
    - Target membership: disable
- Select the godot project in the Project navigator.  
- Then in the File Inspector:
    - Location: Relative to Project
    - Build Rules: Apply Once to a Folder
    - Target Membership: (add your project here)

And of course don't forget to modify your Project-Info.plist as mentioned in the documentation:
 - set "godot_path" to the name of your godot reference folder

Tested on:

- Xcode 16.3
- Minimum iOS 14.0 
- Target iOS device: 18.3.2

**Comment by Grublady** [comment]
Reproducible with macOS 15.2, Xcode 16.3, Godot 4.5-dev2.

Looks pretty thoroughly confirmed, should be OK to remove "needs testing" tag.
CC @AThousandShips

**Comment by MadGeorge** [comment]
Still a valid issue in 4.4.1-stable

**Comment by owlnewworlds** [comment]
> Still a valid issue in 4.4.1-stable

Yes, it's crazy how uselessly complex iOS export is and how it didn't change for years. I've been using Godot for 2 years and a half and the state of iOS Godot is the same : complex and nerves crisis export, outdated and not maintained and incompatible native plugins...

It's even more crazy to know that some devs from Apple joined the contributors for Vision stuff. I wonder how they were able to work with such conditions... or maybe I'm just stupid, but it seems all the mobile Godot developers have issues with iOS.

Sorry, didn't want to be rude or what. The community is amazing, I was just trying to make a new update for iOS and back to errors and crazy stuff :( I really hope one day something will be done regarding iOS. It's so freaking EASY to export to Android !
